src/config.rs

```diff
@@ -6,6 +6,10 @@ pub struct Config {
     pub workers: i32,
     pub proxy: String,
     pub app_version: String,
+    pub environment: Option<String>,
+    pub component: Option<String>,
```

Comment:
This should be configurable on notice level (same with `os`, `hostname`). Ideally, `os` & `hostname` are figured automatically.